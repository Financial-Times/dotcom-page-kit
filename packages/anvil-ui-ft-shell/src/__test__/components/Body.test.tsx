import React from 'react'
import renderer from 'react-test-renderer'
import Subject from '../../components/Body'

describe('anvil-ui-ft-shell/src/components/Body', () => {
  describe('with stringified contents', () => {
    it('renders the given HTML string without escaping', () => {
      const contents = '<p>Hello, World!</p>'
      const tree = renderer.create(<Subject contents={contents} />).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('with component contents', () => {
    const contents = <p>Hello, World!</p>
    const tree = renderer.create(<Subject contents={contents} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  describe('with custom attributes', () => {
    const attributes = { 'data-foo': 'bar', 'data-baz': 'qux' }
    const tree = renderer.create(<Subject contents="" {...attributes} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
