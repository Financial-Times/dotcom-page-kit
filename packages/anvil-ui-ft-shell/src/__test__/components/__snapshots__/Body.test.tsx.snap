// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<body>
  <p>
    Hello, World!
  </p>
</body>
`;

exports[` 2`] = `
<body
  dangerouslySetInnerHTML={
    Object {
      "__html": "",
    }
  }
  data-baz="qux"
  data-foo="bar"
/>
`;

exports[`anvil-ui-ft-shell/src/components/Body with stringified contents renders the given HTML string without escaping 1`] = `
<body
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>Hello, World!</p>",
    }
  }
/>
`;
